### PR TITLE
Add vmware iso support. Fixed #2171

### LINKED ISSUE #2171: VMware Fusion problem
**RancherOS Version: (ros os version)**
 v1.1.1-rc1 
**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
Docker Machine with VMware Fusion driver
VmWare version 8.5.9
macOS 10.13.1

```
➜  ~ docker-machine create --driver vmwarefusion --vmwarefusion-boot2docker-url https://github.com/rancher/os/releases/download/v1.1.1-rc1/rancheros.iso ranchervm
```
It hangs with following output on VM
screen shot 2017-12-09 at 5 19 14 pm

On the terminal I see the following
```
➜  ~ docker-machine create --driver vmwarefusion --vmwarefusion-boot2docker-url https://github.com/rancher/os/releases/download/v1.1.1-rc1/rancheros.iso ranchervm

Running pre-create checks...
(ranchervm) Boot2Docker URL was explicitly set to "https://github.com/rancher/os/releases/download/v1.1.1-rc1/rancheros.iso" at create time, so Docker Machine cannot upgrade this machine to the latest version.
Creating machine...
(ranchervm) Boot2Docker URL was explicitly set to "https://github.com/rancher/os/releases/download/v1.1.1-rc1/rancheros.iso" at create time, so Docker Machine cannot upgrade this machine to the latest version.
(ranchervm) Downloading /Users/mustafa/.docker/machine/cache/boot2docker.iso from https://github.com/rancher/os/releases/download/v1.1.1-rc1/rancheros.iso...
(ranchervm) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(ranchervm) Creating SSH key...
(ranchervm) Creating VM...
(ranchervm) Creating disk '/Users/mustafa/.docker/machine/machines/ranchervm/ranchervm.vmdk'
(ranchervm) Virtual disk creation successful.
(ranchervm) Starting ranchervm...
(ranchervm) Waiting for VM to come online...
```

I also tried https://releases.rancher.com/os/latest/rancheros.iso it gave me the same result.

### PR DESCRIPTION
**Pull request by Jason-ZW**

### CODE REVIEW

File: scripts/images/vmware/Dockerfile.dapper [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+########## General Configuration #####################
+
+ARG DAPPER_HOST_ARCH=amd64
+ARG HOST_ARCH=${DAPPER_HOST_ARCH}
+ARG ARCH=${HOST_ARCH}
+
+ARG OS_REPO=rancher
+ARG HOSTNAME_DEFAULT=rancher
+ARG DISTRIB_ID=RancherOS
+ARG KERNEL_VERSION_amd64=4.15.9-rancher
```
Comment (niusmallnan) [inline scripts/images/vmware/Dockerfile.dapper]: I think we do not need kernel version here

File: scripts/images/vmware/Dockerfile.dapper [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ARG DOCKER_BUILD_VERSION=1.10.3
+ARG DOCKER_BUILD_PATCH_VERSION=v${DOCKER_BUILD_VERSION}-ros1
+
+ARG BUILD_DOCKER_URL_amd64=https://get.docker.com/builds/Linux/x86_64/docker-${DOCKER_BUILD_VERSION}
+ARG BUILD_DOCKER_URL_arm64=https://github.com/rancher/docker/releases/download/${DOCKER_BUILD_PATCH_VERSION}/docker-${DOCKER_BUILD_VERSION}_arm64
```
Comment (niusmallnan) [inline scripts/images/vmware/Dockerfile.dapper]: only support amd64 for vmware, no arm here

File: scripts/images/vmware/Dockerfile.dapper [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# FROM arm64=aarch64/ubuntu:16.04
+
+# get the apt-cacher proxy set
+ARG APTPROXY=
+
+RUN echo "Acquire::http { Proxy \"$APTPROXY\"; };" >> /etc/apt/apt.conf.d/01proxy \
+    && cat /etc/apt/apt.conf.d/01proxy \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        build-essential \
```
Comment (niusmallnan) [inline scripts/images/vmware/Dockerfile.dapper]: It seems like that we do not need all these packages

File: scripts/images/vmware/scripts/build.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+# prepare vmware tools.
+docker pull ${OS_SERVICE_VMTOOL}
+docker save ${OS_SERVICE_VMTOOL} | xz -9 > dist/build/initrd/usr/share/ros/images-vmtools.tar
+
+# package initrd.
+cd dist/build/initrd/
+find | cpio -H newc -o | ${COMPRESS} > ${INITRD}
+
+if \
```
Comment (niusmallnan) [inline scripts/images/vmware/scripts/build.sh]: Do we need a newline here?

File: scripts/images/vmware/Dockerfile.dapper (hunk 1 above)
Comment (Jason-ZW) [reply scripts/images/vmware/Dockerfile.dapper]: BUILD need this arg `vmlinuz-${KERNEL_VERSION}`

File: Makefile [hunk 5]
```diff
@@ -64,6 +64,15 @@ rpi64:
...
 		&& ../../../.dapper
 
+vmware:
+	rm -rf scripts/images/vmware/assets
+	mkdir -p scripts/images/vmware/assets/
+	cp -r dist/ scripts/images/vmware/assets/
+	cp -r build/ scripts/images/vmware/assets/
+	scripts/version > scripts/images/vmware/assets/version.txt
+	cd scripts/images/vmware/ \
+		&& $(shell cat Dockerfile.dapper | grep '^ARG KERNEL_VERSION_amd64=\S\+' | awk -F ' ' '{print $$NF}') $(shell cat Dockerfile.dapper | grep '^ARG DISTRIB_ID=\S\+' | awk -F ' ' '{print $$NF}') ../../../.dapper
```
Comment (niusmallnan) [inline Makefile]: Better code readability:
```
        cd scripts/images/vmware/ \
                && $(shell cat Dockerfile.dapper | grep '^ARG KERNEL_VERSION_amd64=\S\+' | awk -F ' ' '{print $$NF}') \
                      $(shell cat Dockerfile.dapper | grep '^ARG DISTRIB_ID=\S\+' | awk -F ' ' '{print $$NF}') \
                       ../../../.dapper
```

**Pull request merged**